internal/rbd/rbd_util.go

```diff
@@ -335,38 +330,62 @@ func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 	image := pOpts.RbdImageName
 	found, _, err := rbdStatus(ctx, pOpts, cr)
 	if err != nil {
+		klog.Errorf(util.Log(ctx, "failed getting information for image (%s): (%s)"), pOpts.Pool+"/"+pOpts.RbdImageName, err)
+		if strings.Contains(err.Error(), "rbd: error opening image "+image+
+			": (2) No such file or directory") {
+			return ErrImageNotFound{image, err}
+		}
 		return err
 	}
 	if found {
-		klog.Errorf(util.Log(ctx, "rbd %s is still being used"), pOpts)
-		return fmt.Errorf("rbd %s is still being used", pOpts)
+		klog.Errorf(util.Log(ctx, "rbd is still being used "), image)
+		return fmt.Errorf("rbd %s is still being used", image)
 	}
+	// Support deleting the older rbd images whose imageID is not stored in omap
+	if pOpts.ImageID == "" {
+		pOpts.ImageID, err = pOpts.getImageID()
+		if err != nil {
+			return err
+		}
+	}
+	klog.V(4).Infof(util.Log(ctx, "rbd: delete %s using mon %s, pool %s"), image, pOpts.Monitors, pOpts.Pool)
 
-	klog.V(4).Infof(util.Log(ctx, "rbd: rm %s using mon %s"), pOpts, pOpts.Monitors)
+	var ioctx *rados.IOContext
```

Comment:
just use `pOpts.open()` to get the imahe, and use `pOpts.ioctx` after that for the IOContext. No need to destroy the IOContext, as it will be cleaned up by `pOpts.Destroy()` at a later time.

You need to defer closing the image though.

Reply:
would like to avoid opening an image as its not required it.

Reply:
ah, right... Calling librbd from outside a `rbdVolume` method is not very nice, but there is currently no cleaner alternative